test/batch/rules.ts

```diff
@@ -0,0 +1,349 @@
+    "Approved",
+    "Is missing approval",
+    "Has no approval",
+  ] as const) {
+    const setup = function ({
+      setupCoworkers,
+      diff,
+    }: { setupCoworkers?: string[]; diff?: string } = {}) {
+      setupCoworkers ??= coworkers
+      diff ??= condition
+
+      nock(githubWebsite)
+        .get(basePR.diff_url.slice(githubWebsite.length))
+        .reply(200, diff)
+
+      nock(githubApi)
+        .get(reviewsApiPath)
+        .reply(
+          200,
+          scenario === "Approved"
+            ? setupCoworkers.map(function (coworker, id) {
+                return { id, user: { id, login: coworker }, state: "APPROVED" }
+              })
+            : scenario === "Is missing approval"
+            ? [
+                {
+                  id: 1,
+                  user: { id: 1, login: coworkers[0] },
+                  state: "APPROVED",
+                },
+              ]
+            : [],
+        )
+
+      teamMembers = setupCoworkers.map(function (coworker, id) {
+        return { id, login: coworker }
+      })
+      nock(githubApi).get(teamApiPath).reply(200, teamMembers)
+
+      nock(githubApi)
+        .get(changedFilesApiPath)
+        .reply(200, [{ filename: condition }])
+    }
+
+    for (const checkType of ["diff", "changed_files"] as const) {
+      it(`${scenario} on rule including only users for ${checkType}`, async function () {
+        setup()
+
+        nock(githubApi)
+          .get(configFileContentsApiPath)
+          .reply(200, {
+            content: Buffer.from(
+              `
+              rules:
+                - name: ${condition}
+                  condition: ${condition}
+                  check_type: ${checkType}
+                  min_approvals: ${coworkers.length}
+                  users:
+                    - ${coworkers[0]}
+                    - ${coworkers[1]}
+              `,
+            ).toString("base64"),
+          })
+
+        switch (scenario) {
+          case "Has no approval": {
+            nock(githubApi)
+              .post(requestedReviewersApiPath, function (body) {
+                expect(body).toMatchObject({
+                  reviewers: coworkers,
+                  team_reviewers: [],
+                })
+                return true
+              })
+              .reply(201)
+            break
+          }
+          case "Is missing approval": {
+            nock(githubApi)
+              .post(requestedReviewersApiPath, function (body) {
+                expect(body).toMatchObject({
+                  reviewers: [coworkers[1]],
+                  team_reviewers: [],
+                })
+                return true
+              })
+              .reply(201)
+            break
+          }
+        }
+
+        expect(
+          await runChecks(basePR, octokit, logger, {
+            configFilePath,
+            locksReviewTeam: team,
+          }),
+        ).toBe(scenario === "Approved" ? "success" : "failure")
```

Comment:
All the literal `"failure"` and `"success"` in the tests can be refactored to `commitState*` in #44